src/Command/ProbeDispatcherCommand.php

```diff
@@ -320,19 +320,12 @@ private function handleResponse($channel, $data): void
                 break;
 
             case FetchConfiguration::class:
-                $this->logger->info('dispatcher: started handling new configuration response');
                 if (200 === $status) {
-                    $this->logger->info('dispatcher: applying new configuration');
-                    $etag = $response['headers']['etag'];
-                    $this->configuration->updateConfig($contents, $etag);
-                    $this->logger->info(sprintf('dispatcher: new configuration applied (took %.2f seconds)', microtime(true) - $startAt));
-                    $this->logger->info(sprintf('dispatcher: new configuration has %d probes and %d devices', count($this->configuration->getProbes()), $this->configuration->getAllProbesDeviceCount()));
-
-                    $count = 0;
-                    foreach ($this->configuration->getProbes() as $probe) {
-                        $count += ceil($this->configuration->getProbeDeviceCount($probe->getId()) / $this->devicesPerWorker);
-                    }
-                    $this->workerManager->setNumberOfProbeProcesses(intval($count));
+                    $this->configuration = new Configuration($response['headers']['etag'], $contents);
```

Comment:
It's possible that this causes a memory leak if the previous configuration or its content is still referenced.

Reply:
I'm not entirely sure how this would apply, but I tried testing this in the following manner.

![image](https://user-images.githubusercontent.com/22906111/116829739-d798c900-aba5-11eb-8065-dc414ae19824.png)

I let this run for a while to let the `$configurations` property gather some "legacy" configurations. I then printed the array with debug_zval_dump which prints the refcount. Refcount never increases beyond 1.

I added another test afterwards where I unshift a value from the configurations property, dereferencing it everywhere, after which it is garbage collected.

What should I change to avoid the leak you're suggesting, if applicable?

Reply:
I've made your suggested changes to unset the configuration after reloading it. :)